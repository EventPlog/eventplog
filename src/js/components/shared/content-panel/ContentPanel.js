import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  background: white;
  padding: 1rem;
  box-shadow: 1px 2px 4px ${props => lighten(0.45, props.theme.activeLink)};
    
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    border-bottom: 1px solid ${props => lighten(0.2, props.theme.blue)};
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 0.01rem;
    margin: 0 1rem;
    padding-bottom: 5px;
    
    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    margin: 2rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h5 className="content-header">{ title }</h5>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel