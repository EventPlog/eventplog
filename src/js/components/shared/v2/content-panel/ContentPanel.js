import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    margin: 0;
  }

    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    // margin: 2rem 1rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h4 className="content-header">{ title }</h4>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel
