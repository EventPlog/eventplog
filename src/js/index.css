body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

h1, h2, h3, h4, h5, h6 {
  font-family: AndaleMono, "Andale Mono WT", "Lucida Console", "Lucida Sans Typewriter", "DejaVu Sans Mono", "Open Sans";
  color: #a2a3a5;
}

h1, h2, h3 {
  letter-spacing: -1px;
}

h1 {
  font-size: 3rem;
  font-weight: 300;
}

h2 {
  font-size: 2.5rem;
  font-weight: 300;
}

h3 {
  font-size: 2rem;
  font-weight: 500;
}

h4 {
  font-size: 1.8rem;
  font-weight: 500;
}

h5 {
  font-size: 1.5rem;
  font-weight: 500;
}

h6 {
  font-size: 1.2rem;
  font-weight: 500;
}

.app-container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 50px;
}
