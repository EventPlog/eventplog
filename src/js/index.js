import React from 'react';
import ReactDOM from 'react-dom';
import 'semantic-ui-css/semantic.min.css';
import './index.css'
import App from './App.js';
import { BrowserRouter } from 'react-router-dom'
import registerServiceWorker from '../registerServiceWorker';
import { Provider } from 'react-redux'
import configureStore from './store/configureStore'

import { injectGlobal } from 'styled-components';
import { media } from './styles/mixins'
import colors from './styles/theme/variables'

injectGlobal`
  body {
    margin: 0;
    overflow-x: hidden;
	  font-family: 'proxima-nova', sans-serif;
    
  }
  
  .app-container {
    max-width: 1200px;
    margin: 0 auto;
    padding: 0 50px;
    
    ${
      media.phone`
        flex-direction: column;
        padding: 2rem;
      `
    }
  }
  
  *:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
  
  ::selection {
    background: ${ colors.yellow };
  }
  
  ::-moz-selection {
    background: ${ colors.yellow };
  }
  
  p {
    line-height: 1.9rem;
  }
`;


const store = configureStore();

ReactDOM.render(
  <Provider store={store}>
    <BrowserRouter>
      <App store={store} />
    </BrowserRouter>
  </Provider>,
  document.getElementById('root'));

registerServiceWorker();