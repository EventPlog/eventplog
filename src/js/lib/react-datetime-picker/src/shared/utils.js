export {
  min,
  max,
  updateInputWidth,
} from 'react-date-picker/dist/shared/utils';
