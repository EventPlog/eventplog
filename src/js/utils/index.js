export const addHttp = (url) => {
  if (!url.match("~^(?:f|ht)tps?://~i", url)) {
    url = "http://" + url;
  }
  return url;
}

export const validDate = (input) => {
  const _date = new Date(input)
  return _date.getDate().toString() !== 'NaN' ? _date : new Date()
}

export const pluralize = (word, number) => {
  const knownWords = {
    person: 'people',
    follower: 'followers',
    event: 'events'
  }
  return number == 1 ? word : knownWords[word]
}

export const serialize = function(obj) {
  let str = [];
  for (let p in obj)
    if (obj.hasOwnProperty(p)) {
      str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
    }
  return str.join("&");
}

export const paramsToObj = (params) => {
  const obj = {}
  params.split('&').forEach(param => {
    const [key, val] = param.split('=')
    obj[key] = val
  })
  return obj
}

export const getSlugFromHostName = (hostname, knownSubs) => {
  const sub = hostname.substr(0, hostname.indexOf('.'))
  if (!sub) return false
  if(knownSubs.split(',').find(s => s == sub)) return false
  return sub
}

export const genEventLink = (event = {}, community = {}) => {
  if (community && community.id) {
    return `/c/${community.slug || community.id}/e/${event.slug || event.id}`
  }
  return `/e/${event.slug || event.id}`
}

export const genCommunityLink = (community = {}) => (
  community && community.id
  ? `/c/${community.slug || community.id}`
  : ''
)

export const genUserProfileLink = (user = {}) => (
  `/u/${removeSpecialChars(user.less_formal_name)}-${user.id}`
)

export const getUserAvatar = (user = {}) => (
  user.avatar_url || '/sample-avatar.png'
)

export const removeSpecialChars = (str) => (
  str.replace(/[^a-zA-Z\d\-]/g, '').toLowerCase()
)