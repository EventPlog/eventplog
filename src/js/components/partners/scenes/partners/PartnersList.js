import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'
import Pagination from 'js/components/shared/pagination'
import { pluralize, genEventLink } from 'js/utils'
import Presentation from '../partner-card'

const PresentationsList = ({
  title,
  partners,
  attendEvent,
  getPresentations,
  currentUser
}) => {
  const {loading, error, data = [], meta = {}} = partners;
  const shouldDisplayData = (!loading && !error && data);

  return (
    <ContentPanel title={title}>
      {loading && <Loading />}
      {error && <Loading.Error msg={partners.error} />}
      {shouldDisplayData && data.map(partner =>
         <Presentation {...{partner, attendEvent, currentUser}} />
      )}

      {shouldDisplayData && data.length < 1 &&
        <div>
          <p>
            No partner has been added yet.
          </p>
        </div>
      }

      <div className="pagination-wrapper">
      {
        meta && meta.total_pages && (data.length > 0 || meta.current_page > 1)
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getPresentations} />
          : ''
      }
      </div>
    </ContentPanel>
  )
}

export default PresentationsList