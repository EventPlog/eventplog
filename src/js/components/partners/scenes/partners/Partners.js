import React from 'react'
import { Link } from 'react-router-dom'
import styled, {css} from 'styled-components'

import PartnersList from './PartnersList'
import NewPartner from '../new-partner'
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import { genEventLink } from 'js/utils'

const styles = css`
  position: relative;
  
  &.app-container {
    margin-top: 3rem;
    padding: 1rem;
    
    > div {
      width: 100%;
    }
  }
  
  .pic-frame {
    margin: 2rem 0;
  }
  
  .caption {
    margin: 1rem 0;
  }
  
  img {
    max-width: 100%;
  }
  
  .pagination-wrapper {
    display: flex;
  }
`

export const generateTitle = (event) => {
  const title = <Link to={`${genEventLink(event, event.community)}?utm_source=feedback_form`}>{event.title}</Link>
  return <p>All partners - {title}</p>
}

export const Partners = ({
  className,
  partners,
  event,
  loading,
  error,
  currentUser
}) => {

  if (loading) return <Loading />
  if (error) return <Loading.Error msg={error} />

  return (
    <div className={`${className} partners app-container`}>
      <div>
        <PartnersList title={generateTitle(event)}
                  currentUser={currentUser}
                  partners={partners} />
        { event.is_stakeholder &&
          <ContentPanel title="Add a partner">
            <NewPartner newPartner />
          </ContentPanel>
        }
      </div>
    </div>
  )
}

export default styled(Partners)`${styles}`
