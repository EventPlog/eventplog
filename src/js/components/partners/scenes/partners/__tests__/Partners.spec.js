import React from 'react';
import { Partners, generateTitle } from '../Partners';
import PartnersList from '../PartnersList';
import NewPartner from '../../new-partner'
import ContentPanel from 'js/components/shared/content-panel'
import { shallow } from 'enzyme';
import variables from 'js/styles/theme/variables'
import { getUserAvatar } from 'js/utils'

describe('Partners', () => {
  const props = {
    event: {id: 3, is_stakeholder: true},
    presentation: {
      id: 1,
      title: 'An awesome title',
      presentation_type: 'talk',
      summary: 'A short summary',
      details: 'A lot of details',
      user: {id: 1},
      comments: {
        data: []
      }
    },
    theme: variables,
    currentUser: {id: 1},
    handleViewCount: () => {}
  }

  const wrapper = shallow( <Partners {...props} /> );

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

});