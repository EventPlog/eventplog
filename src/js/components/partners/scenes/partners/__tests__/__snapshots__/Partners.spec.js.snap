// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners should render correctly 1`] = `
<div
  className="undefined partners app-container"
>
  <div>
    <PresentationsList
      currentUser={
        Object {
          "id": 1,
        }
      }
      title={
        <p>
          All partners - 
          <Link
            replace={false}
            to="/e/3?utm_source=feedback_form"
          />
        </p>
      }
    />
    <ContentPanel
      title="Add a partner"
    >
      <Component
        newPartner={true}
      />
    </ContentPanel>
  </div>
</div>
`;
